app/models/piece.rb

```diff
@@ -92,9 +94,45 @@ def on_board?(end_point)
 	end
 
 	def do_move!(pos_x, pos_y)
-		self.update_attributes(pos_x: pos_x, pos_y: pos_y)  	
+		self.update_attributes(pos_x: pos_x, pos_y: pos_y)
   end
 
+  def is_occupied?(x, y)
+    game.pieces.where(pos_x: x, pos_y: y, active: true).present?
+  end
+
+  def can_move_without_capture?(start_pos,dest_pos)
+  	start_x, start_y = start_pos
+  	dest_x, dest_y = dest_pos
+    !is_obstructed?([start_x, start_y],[dest_x, dest_y]) && !is_occupied?(dest_x, dest_y) && valid_move?([dest_x, dest_y])
+  end
+
+  def can_move_with_capture?(start_pos,dest_pos)
+		start_x, start_y = start_pos
+  	dest_x, dest_y = dest_pos
+  	dest_piece = game.pieces.find_by(pos_x: dest_x, pos_y: dest_y, active: true)
+  	return (
+  		!is_obstructed?([start_x, start_y],[dest_x, dest_y]) && 
+  		is_occupied?(dest_x, dest_y) &&
+  		#valid_move?([dest_x, dest_y]) &&
+  		dest_piece.color != color 
+  	)
+  end
+
+  def move_to!(start_pos, dest_pos)
+  	dest_x, dest_y = dest_pos
+    if can_move_without_capture?(start_pos, dest_pos)
+        update_attributes(pos_x: dest_x, pos_y: dest_y)
+    elsif can_move_with_capture?(start_pos, dest_pos)
+    		dest_piece = game.pieces.find_by(pos_x: dest_x, pos_y: dest_y, active: true)
+    		dest_piece.update_attributes(active: false)
+    		update_attributes(pos_x: dest_x, pos_y: dest_y)
+    		captured = true
```

Comment:
this is where you assign to the attr_accessor what you are doing here will not allow you to access the captured value in the controller.  You need to get that information back to the controller somehow.